components/BookForm/BookForm.js

```diff
@@ -2,21 +2,24 @@ import Image from "next/image";
 import styled from "styled-components";
 import dummyBook from "../../public/img/dummy/dummy_book.png";
 
-export default function BookForm({ book, onEditSubmit }) {
+export default function BookForm({ book, onEditSubmit, onNewSubmit }) {
   function handleSubmit(event) {
     event.preventDefault();
     const formData = new FormData(event.target);
     const data = Object.fromEntries(formData);
-
-    onEditSubmit(data);
+    if (book) {
+      onEditSubmit(data);
+    } else {
+      onNewSubmit(data);
+    }
```

Comment:
I would not let the component decide which submit function to execute.

I would instead rewrite both handler functions and handle the whole logic in your `_app.js`.
In the locations where you implement your `BookForm` component, you pass either the `handleAddBook` or `handleEditBook` as a prop.
Your BookForm component then only calls the respective handler function when it is submitted.

One function could look like this:
```js
function handleAddBook(event) {
         event.prevenDefault();
         const formData = new FormData(event.target);
         const data = Object.fromEntries(formData);

         const newBook = {
         ...data,
         id: String(bookCounter + 1),
         isActive: true,
         isbn10: "",
         type: "book",
         };

         setBooks([newBook, ...books]);

         router.push(`/`);
}
```

Reply:
Sorry Niklas but I can't agree with that, because the BookForm component knows if data is in the form or not. This should decide what happens with the form data. Two pages access this component.

Reply:
You can leave the part with the FormData within your form component, that is fine.

However, should not decide within your current handleSubmit whether onEditSubmit or onNewSubmit are being called. This decision should take place outside of your form component.

When you pass in those functions as props, once in your `/pages/book/create.js` and once in your `/pages/detail/book/[id]/edit.js`, that is where you make the decision.
In your `/pages/book/create.js` you pass the onNewSubmit function
In your `/pages/detail/book/[id]/edit.js` you pass the onEditSubmit function

Both props should have the same name when being passed, so that in your form component, you only accept one function prop

Reply:
I don't know about this. Can't really tell what is the best way.

Reply:
I have now only given the BookForm the prop onSubmit and forwarded the corresponding prop to the respective pages.

Reply:
perfekt 👍